core-ui/src/shared/useObjectState.ts

```diff
@@ -0,0 +1,20 @@
+import { useState, Dispatch, SetStateAction } from 'react';
+
+export type ObjectStateType<T> = [
+  T,
+  (key: string, value: any) => void,
+  Dispatch<SetStateAction<T>>,
+];
+
+export function useObjectState<T>(): ObjectStateType<T> {
+  const [objectCache, setObjectCache] = useState<T>({} as T);
+
+  const updateObjectCache = (key: string, value: any) => {
```

Comment:
```suggestion
  const updateObjectCache = (key: string, value: T) => {
```

Reply:
nope. because `T` is the main object, while `updateObjectCache` updates only a part of it. not sure if typescript has "one element of an object" type (afaik there is a type for subset, but that's not what's happening here)